src/modules/commander/HealthAndArmingChecks/Common.cpp

```diff
@@ -0,0 +1,227 @@
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "Common.hpp"
+
+void Report::healthFailure(ModeCategory required_modes, HealthComponentIndex component, uint32_t event_id,
+			   const char *message, events::Log log_level)
+{
+	required_modes = applyCurrentNavState(required_modes);
+	healthFailure(required_modes, component, log_level);
+	addEvent(event_id, message, log_level, (uint8_t)required_modes, (uint8_t)component);
+}
+
+void Report::armingCheckFailure(ModeCategory required_modes, HealthComponentIndex component, uint32_t event_id,
+				const char *message, events::Log log_level)
+{
+	required_modes = applyCurrentNavState(required_modes);
+	armingCheckFailure(required_modes, component, log_level);
+	addEvent(event_id, message, log_level, (uint8_t)required_modes, (uint8_t)component);
+}
+
+void Report::setIsPresent(HealthComponentIndex component)
+{
+	HealthResults &health = _results[_current_result].health;
+	health.is_present = health.is_present | (events::common::enums::health_component_t)(1 << (uint8_t)component);
+}
+
+void Report::healthFailure(ModeCategory required_modes, HealthComponentIndex component, events::Log log_level)
+{
+	// update current health results
+	HealthResults &health = _results[_current_result].health;
+
+	if (log_level <= events::Log::Error) {
+		health.error = health.error | (events::common::enums::health_component_t)(1 << (uint8_t)component);
+
+	} else if (log_level <= events::Log::Warning) {
+		health.warning = health.warning | (events::common::enums::health_component_t)(1 << (uint8_t)component);
+	}
+
+	// clear relevant arming bits
+	ModeCategory &can_arm = _results[_current_result].arming_checks.can_arm;
+	can_arm = can_arm & ~required_modes;
+}
+
+void Report::armingCheckFailure(ModeCategory required_modes, HealthComponentIndex component, events::Log log_level)
+{
+	// update current arming check results
+	ArmingCheckResults &arming_checks = _results[_current_result].arming_checks;
+
+	if (log_level <= events::Log::Error) {
+		arming_checks.error = arming_checks.error | (events::common::enums::health_component_t)(1 << (uint8_t)component);
+
+	} else if (log_level <= events::Log::Warning) {
+		arming_checks.warning = arming_checks.warning | (events::common::enums::health_component_t)(1 << (uint8_t)component);
+	}
+
+	// clear relevant arming bits
+	ModeCategory &can_arm = arming_checks.can_arm;
+	can_arm = can_arm & ~required_modes;
+}
+
+void Report::reset(uint8_t nav_state)
+{
+	_current_result = (_current_result + 1) % 2;
+	_results[_current_result].reset();
+	_results[_current_result].nav_state = nav_state;
+	_next_buffer_idx = 0;
+	_buffer_overflowed = false;
+}
+
+ModeCategory Report::applyCurrentNavState(ModeCategory modes) const
+{
+	ModeCategory mode_bit = ModeCategory::None;
```

Comment:
Since it's not a bit

```suggestion
	ModeCategory mode_cat = ModeCategory::None;
```

Reply:
It is a single bit, but if you think it's clearer, I updated it.